### DISCUSSION TITLE
The "on sale" flag should be linked to the reduced prices

### DISCUSSION THREAD

**Discussion by marionf** [PrestaShop/PrestaShop#40164] [Ideas]
#### Is your feature request related to a problem?

Currently, the "On sale!" flag can be added even for products without reduced price, this brings confusion because some users use this flag to highlight products with a reduced price, while others use it to highlight end of stock products without a reduced price.

#### Describe the solution you'd like

Use the "On sale!" flag only for products with a reduced price
And use the ["last chance" flag and page](https://github.com/PrestaShop/PrestaShop/issues/24107) to highlight end of stock products that will be gone after

#### Alternatives you've considered

This issue should be done at the same time or just after this one: https://github.com/PrestaShop/PrestaShop/issues/24107

#### Specs
The checkbox "Display the "On sale!" flag on the product page, and on product listings." is currently in the pricing tab under the "tax rule" field.
We should move at the end of the add specific price form, under "Apply a discount of".
By default, the checkbox should be checked.
When checked, the "On sale!" flag is displayed on the product page and product listing when the specific price conditions are met.
The wording of the checkbox should be renamed as follows:
Label: 'Display the 'On sale' flag' in Admin.Catalog.Feature
Tooltip: 'Use this flag to highlight discounted products. The product will appear on the 'Prices drop' page.'

**Comment by Hlavtox** [comment]
Hi @marionf, in my country, I can show "On sale" without price reduction without any law consequences. What should I do after this issue is implemented? :(

**Comment by marionf** [comment]
Hi @Hlavtox 

The word "on sale", as you can see in [the dictionary](https://www.dictionary.com/browse/on--sale), means that the product is discounted.
So, even if in your country you are allowed to display "on sale" on a product without price reduction, it's a kind of lie as it doesn't feet the definition of "on sale".

Maybe you are confusing with "for sale" which means the product is offered to be sold. And on an online store or even in a physical store, all products are offered to be sold, I don't see the interest of displaying "for sale" on all the products.

**Comment by Hlavtox** [comment]
@marionf Don't forget people can make a discount using catalog rules.

Is it really worth the effort? IMHO, you are just creating problems where they don't exist. Nobody cares about some 'On sale' flag. 🤷🏻‍♂️

**Comment by Hlavtox** [comment]
You want to link this to specific prices right? How will the user know the discount amount when its not shown for specific prices? Isnt that the same if you change the basic price?

You need to do catalog rules for discount to show. How will you check that the conditions are met, when they can be for specific countries and customer groups?

How will you uncheck on_sale if you disable the catalog rule?
